### PR TITLE
ViewCompilation 1.1.0-msbuild3-final does not work with Microsoft.AspNetCore.Mvc 1.1.1

### LINKED ISSUE #78: ViewCompilation 1.1.0-msbuild3-final does not work with Microsoft.AspNetCore.Mvc 1.1.1
Based on comment here - https://github.com/aspnet/MvcPrecompilation/issues/71#issuecomment-277443739

NuGet doesn't look at the `build` directory when determining what binding redirects to produce. Consequently running the exe fails since it can't find the 1.1.0 versions that it's compiled against.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #78

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            .OrderByDescending(f => f)    // On local builds multiple nupkgs are generated.
+            .First();
+        
+        Log.Info("Repacking Nupkg: " + projectNupkg);
+        
+        var extractToDirectory = projectNupkg + "-temp";
+        ZipFile.ExtractToDirectory(projectNupkg, extractToDirectory);
+
+        var runtimesDirectory = Path.Combine(extractToDirectory, "runtimes");
+        var win7x86Directory = Path.Combine(runtimesDirectory, "win7-x86", "lib", "net451");
```
Comment (pranavkm) [inline makefile.shade]: @Eilon - we have two assemblies that have to be used in mutually exclusive contexts (x86 and x64 don't work together). Placing it in the `runtimes` root directory lets NuGet sort this out for us. We did something similar with RazorTooling

File: makefile.shade [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        var net451LibDirectory = Path.Combine(extractToDirectory, "lib", "net451");
+        
+        File.Move(
+            Path.Combine(net451LibDirectory, projectName + ".exe"),
+            Path.Combine(win7x64Directory, projectName + ".exe"));
+        File.Move(
+            Path.Combine(net451LibDirectory, projectName + "-x86.exe"),
+            Path.Combine(win7x86Directory, projectName + "-x86.exe"));
+            
+        File.WriteAllBytes(Path.Combine(net451LibDirectory, "_._"), new byte[0]);
```
Comment (pranavkm) [inline makefile.shade]: Require this to make package target `net451`.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Internal/CompilationOptions.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal
 {
-    public class CompilationOptions
+    internal class CompilationOptions
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Internal/CompilationOptions.cs]: `internal` > Internal namespace

File: testapps/SimpleAppMvc11/SimpleAppMvc11.csproj [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    <OutputType>Exe</OutputType>
+    <RuntimeIdentifier Condition="!$(TargetFramework.StartsWith('netcoreapp'))">win7-x64</RuntimeIdentifier>
+    <MvcRazorCompileOnPublish>true</MvcRazorCompileOnPublish>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.ViewCompilation">
+      <Version>1.1.0-*</Version>
+      <PrivateAssets>All</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.1.1" />
```
Comment (pranavkm) [inline testapps/SimpleAppMvc11/SimpleAppMvc11.csproj]: Mvc 1.1.1

**Comment by Eilon** [comment]
Per email discussion, let's update this to require MVC 1.1.1.

**Comment by pranavkm** [comment]
🆙 📅 - version bump and some fix to the packaging. Package verifier didn't like us calling `ZipFIle.CreateFromDirectory` and none of the available nuget.exes support some of the metadata we have in our packages.

**Comment by Eilon** [comment]
Can we produce a patch build of this called msbuild4-final or similar? (the last one we released was msbuild3-update1, right?)

**Comment by pranavkm** [comment]
We could relabel the current 1.1.0 branch as 1.1.0-msbuild4 so all the fixes since get ported over.

**Pull request merged**